CHANGELOG.md

```diff
@@ -59,6 +59,7 @@
 - Remove `ApplicationPort` and `ApplicationPortData` classes [\#415](https://github.com/eclipse-iceoryx/iceoryx/issues/415)
 - Remove creation pattern from `MemoryMap` and replace it with `MemoryMapBuilder` [\#1036](https://github.com/eclipse-iceoryx/iceoryx/issues/1036)
 - Fix error handling of `TypedUniqueId` and refactor it to `UniquePortId` [\#861](https://github.com/eclipse-iceoryx/iceoryx/issues/861)
+- `cxx::Expects` and `cxx::Ensures` are moved into the header `requires.hpp` [\#996](https://github.com/eclipse-iceoryx/iceoryx/issues/996)
```

Comment:
I rewrote the issue so that it states in more detail what, how and why we are doing here.

@elBoberido but I would leave this line as it is in the changelog? What is your concern here? It states exactly what has changed in the public API.

Reply:
It is just that this PR also adds the functional interface and that it should be part of the chagnelog